pkg/network/state.go

```diff
@@ -570,6 +583,14 @@ func (ns *networkState) storeClosedConnection(c *ConnectionStats) {
 			continue
 		}
 		client.closed.insert(c, ns.maxClosedConns)
+
+		// Check if the buffer is near capacity using the configured threshold
+		if float64(len(client.closed.conns)) >= float64(ns.maxClosedConns)*ns.closedConnectionsBufferThresholdRatio {
+			if !client.closedConnectionsNearCapacity.Load() {
+				log.Warnf("Closed connections buffer for client %s is nearing capacity (%d/%d, threshold: %.2f%%). Setting flag.", clientID, len(client.closed.conns), ns.maxClosedConns, ns.closedConnectionsBufferThresholdRatio*100)
+				client.closedConnectionsNearCapacity.Store(true)
+			}
```

Comment:
isn't it a racy code? two instances can reach to line 589 at the same time

Reply:
I think this code only called when the lock for the state is held, so likely no need to for atomics.